### DISCUSSION TITLE
[crypto-js] crypto.lib.WordArray.create Can Take ArrayBuffer as parameter

### DISCUSSION THREAD

**Discussion by zehuac2** [DefinitelyTyped/DefinitelyTyped#63092] [Issues with a @types package]
Current typing of `crypto-js` reports type error in the following snippet. However, the code runs perfectly.

```ts
import crypto from 'crypto-js';

async function md5(buffer: ArrayBuffer): Promise<string> {
  const words = crypto.lib.WordArray.create(buffer);
  const hash = crypto.MD5(words);
  const base64 = crypto.enc.Base64.stringify(hash);

  return base64;
}
```

[a comment by an automated account is not shown]

**Comment by SevenOutman** [comment]
Hi @Zehua-Chen , there's no signature for `WordArray.create()` that accepts an ArrayBuffer [according to JSDoc from `crypto-js`](https://github.com/brix/crypto-js/blob/4dcaa7afd08f48cd285463b8f9499cdb242605fa/src/core.js#L227-L239). You could convert the buffer to an array before passing it to `WordArray.create()`

**Reply by zehuac2** [reply]
It not only takes ArrayBuffer but also typed arrays like Uint8Array. However, as far as I understand the original documentation on crypto-js now points people to [Forge](https://github.com/digitalbazaar/forge). So I think there might not be a need to adjust this.

**Reply by zehuac2** [reply]
I know it works because I am using it to compute md5 for uploading data to AWS S3. If it doesn't work then AWS will reject my request.

**Reply by SevenOutman** [reply]
I'm sure you know it works. What I meant is that the original documentation does not guarantee it works with ArrayBuffer, so adding this signature would make this @types package not as accurate. In other words, if the official of crypto-js confirms it's designed to work with ArrayBuffer, then it's safer to add this signature declaration.
